service/src/main/java/bio/terra/tanagra/service/artifact/Study.java

```diff
@@ -179,7 +180,8 @@ public Study build() {
       if (properties == null) {
         properties = new HashMap<>();
       }
-      if (id == null) {
+      //true if the id is empty or null
+      if (StringUtils.isEmpty(id)) {
```

Comment:
Added an extra check here for empty, just in case someone decides to pass an empty string as id. Not sure how you may want to handle duplicate ids/data integrity violations, but I'll leave that up to your team.

Reply:
Thanks for the added check.

I think the duplicate ids case should already be handled [on DB insert](https://github.com/DataBiosphere/tanagra/blob/454cc77099f60f88019435e6da807ae341f78d66/service/src/main/java/bio/terra/tanagra/db/StudyDao.java#L77).